test/parallel_api/numeric/numeric.ops/inclusive_scan_by_segment.pass.cpp

```diff
@@ -27,8 +27,82 @@ using namespace oneapi::dpl::execution;
 #endif
 using namespace TestUtils;
 
+//#define DUMP_CHECK_RESULTS
+
+namespace
+{
+#if TEST_DPCPP_BACKEND_PRESENT
+    // accessors
+    template <typename T, int Dim, sycl::access::mode AccMode, sycl::access::target AccTarget,
+              sycl::access::placeholder Placeholder>
+    T*
+    get_host_iterator(sycl::accessor<T, Dim, AccMode, AccTarget, Placeholder>& acc)
+    {
+        return &acc[0];
+    }
+#endif // TEST_DPCPP_BACKEND_PRESENT
+
+    template<typename>
+    struct TVoid
+    {
+        using type = void;
+    };
+
+    template<typename T, typename Type = void>
+    struct HasBeginMethod : std::false_type
+    {
+    };
+
+    template <typename T>
+    struct HasBeginMethod<T, typename TVoid<decltype(&T::begin)>::type> : std::true_type
+    {
+    };
+
+    // containers with begin
+    template <typename Container>
+    auto
+    get_host_iterator(Container& cont) ->
+        typename ::std::enable_if<HasBeginMethod<Container>::value, decltype(cont.begin())>::type
+    {
+        return cont.begin();
+    }
+
+    // pointers
+    template <typename T>
+    T*
+    get_host_iterator(T* data)
+    {
+        return data;
+    }
+
+    // iterators (ramdom access)
+    template <typename Iterator>
+    typename ::std::enable_if<::std::is_same<typename ::std::iterator_traits<Iterator>::iterator_category,
+                                             ::std::random_access_iterator_tag>::value,
+                              Iterator>::type
+    get_host_iterator(Iterator it)
+    {
+        return it;
+    }
+};
+
+template <typename BinaryOperation>
 struct test_inclusive_scan_by_segment
 {
+    template <typename T>
+    struct DifferenceTypeT
+    {
+        using DifferenceType = typename ::std::iterator_traits<T>::difference_type;
+    };
+
+#if TEST_DPCPP_BACKEND_PRESENT
+    template <typename T, int Dim, sycl::access::mode AccMode, sycl::access::target AccTarget, sycl::access::placeholder Placeholder>
+    struct DifferenceTypeT<sycl::accessor<T, Dim, AccMode, AccTarget, Placeholder>>
+    {
+        using DifferenceType = ::std::size_t;
```

Comment:
Usage of `accessor::difference_type` is more preferable here.
(https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#subsec:accessors)

Reply:
Fixed

Reply:
[2022-01-14T07:34:48.843Z] /export/users/sys_bbsycl/UB20_Node_Inner/workspace/OneDPL_CI/UB20_Check/src/test/parallel_api/numeric/numeric.ops/inclusive_scan_by_segment.pass.cpp:102:98: error: no type named 'difference_type' in 'sycl::accessor<unsigned long, 1, sycl::access::mode::read_write, sycl::access::target::host_buffer, sycl::access::placeholder::false_t, sycl::ext::oneapi::accessor_property_list<>>'
[2022-01-14T07:34:48.843Z]         using DifferenceType = typename sycl::accessor<T, Dim, AccMode, AccTarget, Placeholder>::difference_type;
[2022-01-14T07:34:48.843Z]                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~

Reply:
It seems to me that this type not implemented in our used version of compiler and SYCL library.